hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/EntityJoinFromElement.java

```diff
@@ -129,10 +130,22 @@ public JoinFragment toJoinFragment(
 
 			final StringBuilder buffer = new StringBuilder();
```

Comment:
Also, as we discussed on HipChat, please only apply this grouping when the inheritence is joined.